experiments/unbinned_spectrum_dataset/unbinned_datasets.py

```diff
@@ -0,0 +1,193 @@
+import numpy as np
+from astropy.io import fits
+from gammapy.spectrum import (
+    SpectrumDatasetMaker,
+    SpectrumDatasetOnOff,
+    ReflectedRegionsBackgroundMaker,
+)
+from gammapy.utils.regions import list_to_compound_region
+
+
+class UnbinnedSpectrumDatasetOnOff(SpectrumDatasetOnOff):
+    """The UnbinnedSpectrumDatasetOnOff has the events and events_off 
+    attritubutes compared to the SpectrumDatasetOnOff from which it inherits."""
+
+    def __init__(
+        self,
+        models=None,
+        events=None,
+        events_off=None,
+        livetime=None,
+        aeff=None,
+        edisp=None,
+        mask_safe=None,
+        mask_fit=None,
+        acceptance=None,
+        acceptance_off=None,
+        name="",
+        gti=None,
+    ):
+        SpectrumDatasetOnOff.__init__(
+            self,
+            models=models,
+            counts=None,
+            counts_off=None,
+            livetime=livetime,
+            aeff=aeff,
+            edisp=edisp,
+            mask_safe=mask_safe,
+            mask_fit=mask_fit,
+            acceptance=acceptance,
+            acceptance_off=acceptance_off,
+            name=name,
+            gti=gti,
+        )
+        # events in the ON and OFF regions
+        self.events = events
+        self.events_off = events_off
+
+    def write(self, datapath, overwrite):
+        """write the ON / OFF event lists and the IRFs in a single HDUList"""
```

Comment:
Why not just use `hdu = fits.BinTableHDU(self.events.table)`? And `hdu_off = fits.BinTableHDU(self.events_off.table)`?